airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java

```diff
@@ -35,30 +44,75 @@ public ScheduleRetrieverOutput getTimeToWait(final ScheduleRetrieverInput input)
     try {
       final StandardSync standardSync = configRepository.getStandardSync(input.getConnectionId());
 
-      if (standardSync.getSchedule() == null || standardSync.getStatus() != Status.ACTIVE) {
-        // Manual syncs wait for their first run
-        return new ScheduleRetrieverOutput(Duration.ofDays(100 * 365));
+      if (standardSync.getScheduleType() != null) {
+        return this.getTimeToWaitFromScheduleType(standardSync, input.getConnectionId());
       }
+      return this.getTimeToWaitFromLegacy(standardSync, input.getConnectionId());
+    } catch (final IOException | JsonValidationException | ConfigNotFoundException e) {
+      throw new RetryableException(e);
+    }
+  }
 
-      final Optional<Job> previousJobOptional = jobPersistence.getLastReplicationJob(input.getConnectionId());
+  private ScheduleRetrieverOutput getTimeToWaitFromScheduleType(final StandardSync standardSync, UUID connectionId) throws IOException {
+    if (standardSync.getScheduleType() == ScheduleType.MANUAL || standardSync.getStatus() != Status.ACTIVE) {
+      // Manual syncs wait for their first run
+      return new ScheduleRetrieverOutput(Duration.ofDays(100 * 365));
+    }
+
+    final Optional<Job> previousJobOptional = jobPersistence.getLastReplicationJob(connectionId);
 
-      if (previousJobOptional.isEmpty() && standardSync.getSchedule() != null) {
-        // Non-manual syncs don't wait for their first run
+    if (standardSync.getScheduleType() == ScheduleType.BASIC_SCHEDULE) {
+      if (previousJobOptional.isEmpty()) {
+        // Basic schedules don't wait for their first run.
         return new ScheduleRetrieverOutput(Duration.ZERO);
       }
-
       final Job previousJob = previousJobOptional.get();
       final long prevRunStart = previousJob.getStartedAtInSecond().orElse(previousJob.getCreatedAtInSecond());
-
-      final long nextRunStart = prevRunStart + ScheduleHelpers.getIntervalInSecond(standardSync.getSchedule());
-
+      final long nextRunStart = prevRunStart + ScheduleHelpers.getIntervalInSecond(standardSync.getSchedule().getBasicSchedule());
       final Duration timeToWait = Duration.ofSeconds(
           Math.max(0, nextRunStart - currentSecondsSupplier.get()));
-
       return new ScheduleRetrieverOutput(timeToWait);
-    } catch (final IOException | JsonValidationException | ConfigNotFoundException e) {
-      throw new RetryableException(e);
     }
+
+    else { // standardSync.getScheduleType() == ScheduleType.CRON
+      final Cron scheduleCron = standardSync.getSchedule().getCron();
+      final TimeZone timeZone = DateTimeZone.forID(scheduleCron.getCronTimeZone()).toTimeZone();
+      try {
+        final CronExpression cronExpression = new CronExpression(scheduleCron.getCronExpression());
+        cronExpression.setTimeZone(timeZone);
```

Comment:
a note for the future: 
here we assume we are able to parse the cronstring. this is fine as long as we make sure we validate this at the api layer. since this will be done in a follow up issue, leaving a comment to make sure we don't forget

Reply:
+1, was a bit unsure the best way to handle this. In this code, this'll throw a `ParseException` which will get re-thrown as a `DateTimeException` which is `used to indicate a problem while calculating a date-time`.

Validating in the API layer certainly seems right.